src/azure-cli/azure/cli/command_modules/aro/_params.py

```diff
@@ -0,0 +1,91 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azure.cli.command_modules.aro._validators import validate_cidr
+from azure.cli.command_modules.aro._validators import validate_client_id
+from azure.cli.command_modules.aro._validators import validate_client_secret
+from azure.cli.command_modules.aro._validators import validate_cluster_resource_group
+from azure.cli.command_modules.aro._validators import validate_domain
+from azure.cli.command_modules.aro._validators import validate_pull_secret
+from azure.cli.command_modules.aro._validators import validate_subnet
+from azure.cli.command_modules.aro._validators import validate_visibility
+from azure.cli.command_modules.aro._validators import validate_vnet
+from azure.cli.command_modules.aro._validators import validate_vnet_resource_group_name
+from azure.cli.command_modules.aro._validators import validate_worker_count
+from azure.cli.command_modules.aro._validators import validate_worker_vm_disk_size_gb
+from azure.cli.core.commands.parameters import name_type
+from azure.cli.core.commands.parameters import resource_group_name_type
+from azure.cli.core.commands.parameters import tags_type
+from azure.cli.core.commands.validators import get_default_location_from_resource_group
+
+
+def load_arguments(self, _):
+    with self.argument_context('aro') as c:
+        c.argument('location',
+                   validator=get_default_location_from_resource_group)
+        c.argument('resource_name',
+                   name_type,
+                   help='Name of cluster.')
+        c.argument('tags',
+                   tags_type)
+
+        c.argument('pull_secret',
+                   help='Pull secret of cluster.',
+                   validator=validate_pull_secret)
+        c.argument('domain',
+                   help='Domain of cluster.',
+                   validator=validate_domain)
+        c.argument('cluster_resource_group',
+                   help='Resource group of cluster.',
```

Comment:
Why there is anohter resource group for cluster? Is it the same as the gloabl defaule resource group?

Reply:
It is not. `resource_group` will contain cluster objects of its own , where `cluster_resource_group` is where the cluster resource components are deployed (if not provided we will create one for the user, but the user can override its name using this parameter and it will be locked for editing). This will be covered by user facing documentation, and it is the nature of the service architecture.